src/flags/index.js

```diff
@@ -20,7 +20,7 @@ export type Flag <T> = {
   completion?: ?{
     cacheDuration?: ?number,
     cacheKey?: ?string,
-    options: () => Promise<string[]>
+    options: (?any) => Promise<string[]>
```

Comment:
The command should definitely not be passed in. The flag should provide the same options no matter what command it is being called on. Otherwise we would have to cache each command's apps separately.

In this case I think you need an instantiated Output object don't you? That should be all that's necessary.